### PR TITLE
Remove the `extern`ed `StatBag` from `ws-auth`

### PR DESCRIPTION
**Pull request by fredmorcos**
### Short description
This removes the `extern StatBag` object from `ws-auth` and instead passes it around as a function/method argument. We should do this, in general, as part of a wider cleanup.

This also cleans up and formats some files automatically using `clang-format`.

Better reviewed on a per-commit basis.

### Checklist
I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [X] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by Habbie** [comment]
This may require a rebase after #13153 which I hope to merge soon.

**Comment by Habbie** [comment]
This indeed requires a rebase now

**Comment by fredmorcos** [comment]
Rebased and cleaned up `ws-auth.cc` and `ws-api.cc`

**Review by chbruyand — APPROVED** [review]
LGTM!

File: pdns/ws-auth.cc [hunk 1]
```diff
@@ -896,19 +901,17 @@ static bool isValidMetadataKind(const string& kind, bool readonly) {
...
+  static vector<string> protectedOptions{
     "API-RECTIFY",
     "AXFR-MASTER-TSIG",
     "NSEC3NARROW",
     "NSEC3PARAM",
     "PRESIGNED",
     "LUA-AXFR-SCRIPT",
-    "TSIG-ALLOW-AXFR"
-  };
+    "TSIG-ALLOW-AXFR"};
```
Comment (Habbie) [inline pdns/ws-auth.cc:973]: this formatting surprises me - it reduces the quality of diffs

but, that's not a discussion for here on this PR I think

**Review by Habbie — COMMENTED** [review]
I was mid-review when you merged (which is fine!). Submitting my one comment anyway :)

File: pdns/ws-auth.cc (hunk 1 above)
Comment (fredmorcos) [reply pdns/ws-auth.cc:973]: I fixed it in #13663

**Pull request merged**